app/controllers/profile_controller.rb

```diff
@@ -18,4 +18,27 @@ def show
     @page_subtitle = ts("%{username} - Profile", username: @user.login)
   end
 
+  def pseuds
+    respond_to do |format|
+      format.html do
+        redirect_to user_pseuds_path(@user)
+      end
+
+      format.js
+    end
+  end
+
+  private
+
+  def load_user_and_pseuds
+    @user = User.find_by(login: params[:user_id])
+
+    if @user.nil?
+      flash[:error] = ts("Sorry, there's no user by that name.")
+      redirect_to root_path
+      return
+    end
```

Comment:
Based on recent behaviour changes for misspelled usernames ([#4312](https://github.com/otwcode/otwarchive/pull/4312#pullrequestreview-1162799688)), I think this should use `find_by!` so it results in a 404 error instead of redirecting when the user is nonexistent.